tests/functional/Combobox.js

```diff
@@ -857,6 +857,151 @@ define([
+				assert.isNull(navigatedDescendant, "navigatedDescendant should be still null.");
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^France/.test(navigatedDescendant),
+					"navigatedDescendant after first ARROW_DOWN: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN) // focus on Cananda
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^Canada/.test(navigatedDescendant),
+					"navigatedDescendant after forth ARROW_DOWN: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ESCAPE)
+			.sleep(500) // wait for async closing
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				// note: even if the popup it's closed, list should maintain its navigatedDescendant set.
+				assert(/^Canada/.test(navigatedDescendant),
+					"navigatedDescendant after closing the popup: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.sleep(500) // wait for async opening
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^Canada/.test(navigatedDescendant),
+					"navigatedDescendant after opening the popup: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN) // focus on Cananda
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^China/.test(navigatedDescendant),
+					"navigatedDescendant after two ARROW_DOWN: " + navigatedDescendant);
+			});
+	};
+
+	var checkNavigatedDescendantWithSelection = function (remote, comboId) {
+		var executeExpr = "return getNavigatedDescendant(\"" + comboId + "\");";
+		return loadFile(remote, "./Combobox-decl.html")
+			.execute(comboId + ".focus();")
+			.pressKeys(keys.ARROW_DOWN)
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert.isNull(navigatedDescendant, "navigatedDescendant should be still null.");
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^France/.test(navigatedDescendant),
+					"navigatedDescendant after first ARROW_DOWN: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN)
+			.findById(comboId + "_item2")
+			.click() // UK selected
+			.end()
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^UK/.test(navigatedDescendant),
+					"navigatedDescendant after two ARROW_DOWN: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN)
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^Canada/.test(navigatedDescendant),
+					"navigatedDescendant after other two ARROW_DOWN: " + navigatedDescendant);
+			})
```

Comment:
I'm confused.  The user clicks "UK" to select it, which causes the dropdown to close, and then the user re-opens the dropdown... shouldn't the focus be on "UK"?

On another note, this test keeps checking `this.list.navigatedDescendant`, but especially since this is a functional test, it seems like what you should really be testing for is where the focus goes, or for aria attributes like aria-selected, or something else that directly affects the user.